code/src/pages/products/ProductsPage.tsx

```diff
@@ -1,22 +1,44 @@
+import { useSearchParams } from "react-router-dom";
+
 import AppTypography from "@/components/app-typography/AppTypography";
 import AppBox from "@/components/app-box/AppBox";
 import ProductCard from "@/components/product-card/ProductCard";
 import PageWrapper from "@/layouts/page-wrapper/PageWrapper";
 import { useGetProductsQuery } from "@/store/api/productsApi";
 
+import AppPagination from "@/components/app-pagination/AppPagination";
+import AppContainer from "@/components/app-container/AppContainer";
+
 import "@/pages/products/ProductsPage.scss";
 
 const ProductsPage = () => {
-  const { data: products, isLoading } = useGetProductsQuery();
+  const [searchParams] = useSearchParams();
+
+  const searchParamsPage = searchParams.get("page");
+  const isPageValid = searchParamsPage && !Number.isNaN(searchParamsPage);
+  const page = isPageValid ? Number(searchParamsPage) : 1;
+
+  const { data: productsData, isLoading } = useGetProductsQuery({
+    page,
+    itemsPerPage: 10
+  });
 
-  //@TODO Create Skeleton for product items
   if (isLoading) return <AppTypography>Loading...</AppTypography>;
 
-  const productCards = products
-    ? products
-        .slice(0, 4)
-        .map((product) => <ProductCard key={product.id} product={product} />)
-    : null;
+  const pagesCount = productsData?.pagesCount;
+
+  //@TODO Create Skeleton for product items
+  const productCards =
+    Boolean(productsData?.items) &&
```

Comment:
Why are you using `Boolean` here? I think it's redundant. Also, it would be better to rename it to `products`

Reply:
Thanks for noticing. I removed the checking because it is redundant.